prac_07/project_management.py

```diff
@@ -0,0 +1,199 @@
+    while choice != "Q":
+        if choice == "L":
+            load_project(data)
+        elif choice == "S":
+            save_project(data)
+        elif choice == "D":
+            display_projects(data)
+        elif choice == "F":
+            filter_project(data)
+        elif choice == "A":
+            add_new_project(data)
+        elif choice == "U":
+            update_project(data)
+        else:
+            print("Invalid choice")
+        print(MENU)
+        choice = input(">>> ").upper()
+    save_data(data, FILENAME)
+    print("Thank you for using custom-built project management software.")
+
+
+def save_project(data):
+    """Save project."""
+    new_saving_filename = input("New saving filename: ")
+    save_data(data, new_saving_filename)
+
+
+def load_project(data):
+    """Load project."""
+    new_loading_filename = input("New filename:")
+    load_data(data, new_loading_filename)
+
+
+def filter_project(data):
+    """Filter project by date."""
+    filtered_projects = []
+    filter_date = input("Show projects that start after date (dd/mm/yy): ")
+    filter_date = datetime.datetime.strptime(filter_date, "%d/%m/%Y").date()
+    for project in data:
+        project.start_date = datetime.datetime.strptime(project.start_date, "%d/%m/%Y").date()
+        if filter_date <= project.start_date:
+            filtered_projects.append(project)
+        else:
+            project.start_date = project.start_date.strftime("%d/%m/%Y")
+    filtered_projects.sort(key=attrgetter("start_date"))
+    for filtered_project in filtered_projects:
+        filtered_project.start_date = filtered_project.start_date.strftime("%d/%m/%Y")
+        print(filtered_project)
+
+
+def update_project(data):
+    """Update project."""
+    for i, new_project in enumerate(data):
+        print(f"{i} {new_project}")
+    project_choice = validate_detail("Project choice: ", int, MINIMUM_PROJECT_CHOICE, len(data) - 1)
+    print(data[project_choice])
+    update_percentage(data, project_choice)
+    update_priority(data, project_choice)
+
+
+def update_priority(data, project_choice):
+    """Update priority."""
+    is_valid_input = False
+    while not is_valid_input:
+        try:
+            new_priority = input("New Priority: ")
+            if new_priority != "":
+                if int(new_priority) < MINIMUM_PRIORITY or int(new_priority) > MAXIMUM_PRIORITY:
+                    print("Error priority.")
+                else:
+                    data[project_choice].priority = int(new_priority)
+                    is_valid_input = True
+            else:
+                is_valid_input = True
+        except ValueError:
+            print("Priority should be a number.")
+
+
+def update_percentage(data, project_choice):
+    """Update percentage."""
+    is_valid_input = False
+    while not is_valid_input:
+        try:
+            new_percentage = input("New Percentage: ")
+            if new_percentage != "":
```

Comment:
Instead of getting new_percentage as a string and converting it to an int, why not get it as an int the first time and ensure it's a number with a get valid number function?